lib/Model/Dao/AbstractDao.php

```diff
@@ -95,4 +95,13 @@ public function resetValidTableColumnsCache($table)
         }
         Cache::clearTags(['system', 'resource']);
     }
+
+    public static function getForeignKeyName($table, $column) {
+        $fkName = 'fk_'.$table.'__'.$column;
+        if (strlen($fkName) > 64) {
+            $fkName = 'fk_'.md5($fkName);
```

Comment:
```suggestion
            $fkName = substr($fkName, 0, 55) . '_' . hash('crc32', $fkName);
```
I think keeping a prefix here would be nice and `crc32` should be enough randomness.